### ISSUE TITLE
Comments/KDoc indentation not checked

### ISSUE THREAD

**Issue by jfresen** [pinterest/ktlint#297]
In the following code, only the last two errors in indentation are detected:
```
fun foo() {               
    /**
       * not detected
     */
    /*
       * not detected
     */
    //
      // detected
    //
    var foo = 1
      foo += 1 // detected
    foo -= 1
}
```

**Comment by shyiko** [comment]
`ktlint -F --experimental` (since [ktlint@0.31.0](https://github.com/shyiko/ktlint/blob/master/CHANGELOG.md#0310---2019-03-10)) produces

```kotlin
fun foo() {
    /**
     * not detected
     */
    
    /*
       * not detected
     */
    //
    // detected
    //
    var foo = 1
    foo += 1 // detected
    foo -= 1
}
```

`/**/` content is not checked because it's essentially freeform. I suppose we can check whether all !blank lines start with * and align them if they are but [it's not implemented yet](https://github.com/shyiko/ktlint/blob/fb31f9f4971b359885d9d4b2acabcbf7a44116c3/ktlint-ruleset-experimental/src/main/kotlin/com/github/shyiko/ktlint/ruleset/experimental/IndentationRule.kt#L935).

**Comment by Monchi** [comment]
@shyiko You mean the code below should be allowed?
```kotlin
fun foo() {
    /*
            * comment
     */
}
```

In my opinion, the text in comment should be freeform, but starting `*` should be aligned.

```kotlin
```kotlin
fun foo() {
    /*
     * ugly
     *         indent
     *    in comment
     * should be allowed
     */
    /*
     * ugly
           * indent
       * of comment head
     * should not be allowed
     */
}
```
```

**Comment by Tapchicoma** [comment]
I would agree that `*` should be aligned and have whitespace after, then text could be free-form.

**Comment by paul-dingemans** [comment]
The issue still (partly) exists in current 0.44.0 version. The KDoc and EOL comments are fixed. The incorrectly indented block comment is still undetected. Original example is now formatted as:
```
fun foo() {
    /**
     * not detected
     */
    val bar = 1
    /*
       * not detected
     */
    //
    // detected
    //
    var foo = 1
    foo += 1 // detected
    foo -= 1
}
```

**Comment by matthewadams** [comment]
It'd be nice if this also hard-wrapped KDoc (Javadoc) lines at some margin, like 80.

**Comment by paul-dingemans** [comment]
> It'd be nice if this also hard-wrapped KDoc (Javadoc) lines at some standard margin.
> 
> So that, with an 80-character margin for example,
> 
> ```
> /**
>  * This is a really really really really really really really really really really really really really really really really really long line.
>  */
> ```
> 
> is reformatted to
> 
> ```
> /**
>  * This is a really really really really really really really really really
>  * really really really really really really really really long line.
>  */
> ```

This is not related to the original issue. Also, I don't think that KDOC should be wrapped. If you really would like to have such functionality it is best to write a custom rule for it. See https://pinterest.github.io/ktlint/api/custom-rule-set/ for a starting point to create a custom rule.
